src/main/java/slimeknights/tconstruct/common/data/TConstructRecipeProvider.java

```diff
@@ -685,6 +675,92 @@ private void addSmelteryRecipes(Consumer<IFinishedRecipe> consumer) {
       .patternLine(" # ")
       .addCriterion("has_item", this.hasItem(TinkerSmeltery.searedBrick))
       .build(consumer, location("smeltery/faucet"));
+
+    ShapedRecipeBuilder.shapedRecipe(TinkerSmeltery.castingBasin.get())
+      .key('#', TinkerSmeltery.searedBrick)
+      .patternLine("# #")
+      .patternLine("# #")
+      .patternLine("###")
+      .addCriterion("has_item", this.hasItem(TinkerSmeltery.searedBrick))
+      .build(consumer, location("smeltery/casting/basin"));
+
+    ShapedRecipeBuilder.shapedRecipe(TinkerSmeltery.castingTable.get())
+      .key('#', TinkerSmeltery.searedBrick)
+      .patternLine("###")
+      .patternLine("# #")
+      .patternLine("# #")
+      .addCriterion("has_item", this.hasItem(TinkerSmeltery.searedBrick))
+      .build(consumer, location("smeltery/casting/table"));
+  }
+
+  private void addCastingRecipes(Consumer<IFinishedRecipe> consumer) {
+    // Pure Fluid Recipes
+    // Slime
+    this.addSlimeCastingRecipe(consumer, TinkerFluids.blood, SlimeType.BLOOD);
+    this.addSlimeCastingRecipe(consumer, TinkerFluids.blueSlime, SlimeType.BLUE);
+    this.addSlimeCastingRecipe(consumer, TinkerFluids.purpleSlime, SlimeType.PURPLE);
+    this.addBucketCastingRecipe(consumer, TinkerFluids.searedStone);
```

Comment:
I don't like a separate recipe for each bucket, that will make mod compat a pain. Make a dynamic fluid container recipe instead. Will accept `Items.BUCKET` and return the bucket filled with the relevant fluid. To make it work with the recipe, either extend the casting table recipe, or have a shared base class.

Reply:
Honestly unsure how that would go down. Are our buckets of molten stuff
just instances of the vanilla bucket? Or the forge bucket?

Reply:
Does not matter, IItemFluidHandler. Look into Forge's fluid utils to fill an item fluid container. So basically, the recipe will fill the input stack (passed in as iron bucket in JSON) with the 100b of fluid and get the result. Look at the 1.12 recipe for an example

Reply:
I see... in 1.12 the casting recipe setup was an interface instead of an abstract class.
How's about this, we get this pr merged more/less as-is and I'll work out the details
for this setup in 1.16.x, I've already racked my brain quite enough over it, I think.

Reply:
Sure, we can deal with the bucket recipe afterwards